javagen/src/main/java/com/azure/autorest/mapper/ServiceClientMapper.java

```diff
@@ -108,14 +106,14 @@ public ServiceClient map(CodeModel codeModel) {
                 serviceClientPropertyClientType = serviceClientPropertyClientType.asNullable();
             }
 
-            boolean serviceClientPropertyIsReadOnly = p.getSchema() instanceof ConstantSchema;
```

Comment:
Is it possible for you to add a `isFluent` guard for old logic? "api-version" is a Constant in m4, hence it should be not modifiable by definition.

IMHO, if one want "api-version" to be settable, it would more like an exception for one single property (instead of making every Constant property non-constant).

And Jonathan asked us to make it not settable in mgmt-plane.

Reply:
I know previously it is settable, but that is due to a bug:

```
boolean serviceClientPropertyIsReadOnly = serviceClientPropertyClientType instanceof ConstantSchema;
```

`serviceClientPropertyClientType` is client model type, `ConstantSchema` is code model type (m4). They will never match.